creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1194,38 +1195,40 @@ class SnapbackSM {
    * @notice this will issue sync to healthy secondary and update replica set away from unhealthy secondary
    */
   async aggregateReconfigAndPotentialSyncOps(nodeUsers, unhealthyPeers) {
-    let requiredUpdateReplicaSetOps = []
-    let potentialSyncRequests = []
-
-    for (const nodeUser of nodeUsers) {
-      const { primarySpID, secondary1SpID, secondary2SpID } = nodeUser
-
-      // If these spIds are undefined, it means that the Discovery Node does not have the latest code to return
-      // these fields. Fallback to the original aggregation logic. If present, use the spIds to determine whether
-      // reconfigs/syncs are necessary.
-      let subsetReplicaOps, subsetSyncReqs
-      if (primarySpID && secondary1SpID && secondary2SpID) {
-        ;({
-          requiredUpdateReplicaSetOps: subsetReplicaOps,
-          potentialSyncRequests: subsetSyncReqs
-        } = await this._aggregateOpsWithQueriedSpIds(nodeUser, unhealthyPeers))
-      } else {
-        // TODO: remove this else case once all the Discovery Nodes have upgraded and return the necessary replica set spIds
-        ;({
-          requiredUpdateReplicaSetOps: subsetReplicaOps,
-          potentialSyncRequests: subsetSyncReqs
-        } = await this._aggregateOpsWithoutQueriedSpIds(
-          nodeUser,
-          unhealthyPeers
-        ))
-      }
-
-      requiredUpdateReplicaSetOps =
-        requiredUpdateReplicaSetOps.concat(subsetReplicaOps)
-      potentialSyncRequests = potentialSyncRequests.concat(subsetSyncReqs)
-    }
-
-    return { requiredUpdateReplicaSetOps, potentialSyncRequests }
+    const _aggregateOpsWithQueriedSpIds =
+      this._aggregateOpsWithQueriedSpIds.bind(this)
+    return Promise.all(
+      nodeUsers.map((nodeUser) =>
+        _aggregateOpsWithQueriedSpIds(nodeUser, unhealthyPeers)
+      )
+    )
+      .then((results) => {
+        const { requiredUpdateReplicaSetOps, potentialSyncRequests } =
+          results.reduce((result1, result2) => {
```

Comment:
can we make result1 and result2 more informative? like `previousNodeUser` and `currentNodeUser`